src/test/netbase_tests.cpp

```diff
@@ -477,11 +475,14 @@ BOOST_AUTO_TEST_CASE(netpermissions_test)
 
 BOOST_AUTO_TEST_CASE(netbase_dont_resolve_strings_with_embedded_nul_characters)
 {
-    CNetAddr addr;
-    BOOST_CHECK(LookupHost("127.0.0.1"s, addr, false));
-    BOOST_CHECK(!LookupHost("127.0.0.1\0"s, addr, false));
-    BOOST_CHECK(!LookupHost("127.0.0.1\0example.com"s, addr, false));
-    BOOST_CHECK(!LookupHost("127.0.0.1\0example.com\0"s, addr, false));
+    std::optional<CNetAddr> addr{LookupHost("127.0.0.1"s, false)};
+    BOOST_CHECK(addr.has_value());
+    addr = LookupHost("127.0.0.1\0"s, false);
+    BOOST_CHECK(!addr.has_value());
+    addr = LookupHost("127.0.0.1\0example.com"s, false);
+    BOOST_CHECK(!addr.has_value());
+    addr = LookupHost("127.0.0.1\0example.com\0"s, false);
+    BOOST_CHECK(!addr.has_value());
```

Comment:
This can be shorter:

```cpp
    BOOST_CHECK(LookupHost("127.0.0.1"s, false).has_value());
    BOOST_CHECK(!LookupHost("127.0.0.1\0"s, false).has_value());
    BOOST_CHECK(!LookupHost("127.0.0.1\0example.com"s, false).has_value());
    BOOST_CHECK(!LookupHost("127.0.0.1\0example.com\0"s, false).has_value());
```

Reply:
Much more cleaner, nice!